test/coverage-test/test/run-dynamic-coverage.test.ts

```diff
@@ -0,0 +1,62 @@
+import { expect } from 'vitest'
+import { cleanupCoverageJson, readCoverageMap, runVitest, test } from '../utils'
+
+test('enableCoverage() collects coverage after being called', async () => {
+  await cleanupCoverageJson()
+
+  // Run a minimal suite where coverage starts disabled, then enable it and rerun.
+  const { ctx } = await runVitest({
+    include: ['fixtures/test/math.test.ts'],
+    coverage: {
+      // start disabled and turn on dynamically
+      enabled: false,
+      reporter: 'json',
+      include: [
+        'fixtures/src/math.ts',
+        'fixtures/src/untested-file.ts',
+      ],
+    },
+  })
+
+  await expect(readCoverageMap(), 'coverage map is not on the disk').rejects.toThrowError(/no such file/)
```

Comment:
Should this custom error message be the opposite? When `readCoverageMap` does not throw `no such file`, we would want to print something like `coverage map was written to file system when expected it not to etc`.

Reply:
I usually use this field to describe what `expect` expects